src/coreclr/jit/emitxarch.cpp

```diff
@@ -1969,14 +1994,14 @@ bool IsXMMReg(regNumber reg)
 unsigned HighAwareRegEncoding(regNumber reg)
 {
     static_assert((REG_XMM0 & 0x7) == 0, "bad XMMBASE");
-    return (unsigned)(reg & 0xF);
+    return (unsigned)(AbsRegNumber(reg) & 0xF);
```

Comment:
given that this will never see SIMD register, will this method get mask registers? If not, we can directly encode the GPR. If it just gets GPR and mask, then may be we might need version of `AbsRegNumber()` that just checks if it is GPR or mask?

Reply:
Actually, this will get SIMD registers. `insEncodeReg3456` uses this helper and it is used to encode SIMD registers. it's not relevant for mask registers I think